pkg/numaresourcesscheduler/objectstate/sched/sched.go

```diff
@@ -205,13 +207,20 @@ func UpdateSchedulerName(cm *corev1.ConfigMap, name string) error {
 		return err
 	}
 
-	for i, schedProf := range schedCfg.Profiles {
+	for i := range schedCfg.Profiles {
+		schedProf := &schedCfg.Profiles[i]
 		// if we have a configuration for the NodeResourceTopologyMatch
 		// this is a valid profile
-		for _, plugin := range schedProf.PluginConfig {
-			if plugin.Name == SchedulerPluginName {
-				schedCfg.Profiles[i].SchedulerName = &name
+		for j := range schedProf.PluginConfig {
+			plugin := &schedProf.PluginConfig[j]
+			if plugin.Name != SchedulerPluginName {
+				continue
 			}
+			schedName := name
+			if schedProf.SchedulerName != nil && strings.HasSuffix(*schedProf.SchedulerName, SchedulerProfileSortedSuffix) {
+				schedName = name + SchedulerProfileSortedSuffix
+			}
+			schedProf.SchedulerName = &schedName
```

Comment:
SchedulerName is a pointer ?

Reply:
According to the doc, it will be set to the default scheduler in case it's empty.